ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/worldview/BonsaiWorldState.java

```diff
@@ -517,7 +541,10 @@ public Hash frontierRootHash() {
     return calculateRootHash(
         Optional.of(
             new BonsaiWorldStateKeyValueStorage.Updater(
-                noOpSegmentedTx, noOpTx, worldStateStorage.getFlatDbStrategy())),
+                noOpSegmentedTx,
+                noOpTx,
+                worldStateStorage.getFlatDbStrategy(),
+                worldStateStorage.getWorldStateStorage())),
```

Comment:
We have CodeStorageStrategy, why do we need to expose composed worldstate storage ? in FlatDbStrategy we are not even using this param and instead use codeStorageStrategy.  

If we get rid of this unnecessary usage, we can remove the accessor from Bonsai WK KV storage class.